ndc-test/src/lib.rs

```diff
@@ -141,37 +141,47 @@ pub async fn test_snapshots_in_directory_with<
     snapshots_dir: PathBuf,
     f: impl Fn(Req) -> F,
 ) {
-    match std::fs::read_dir(snapshots_dir) {
+    match read_dir_sorted_by_name(snapshots_dir) {
         Ok(dir) => {
             for entry in dir {
-                let entry = entry.expect("Error reading snapshot directory entry");
-
-                test!(
-                    entry.file_name().to_str().unwrap_or("{unknown}"),
-                    reporter,
-                    {
-                        async {
-                            let path = entry.path();
+                if entry.metadata().is_ok_and(|md| md.is_dir()) {
+                    test!(
+                        entry.file_name().to_str().unwrap_or("{unknown}"),
+                        reporter,
+                        {
+                            async {
+                                let path = entry.path();
 
-                            let snapshot_pathbuf = path.join("expected.json");
-                            let snapshot_path = snapshot_pathbuf.as_path();
+                                let snapshot_pathbuf = path.join("expected.json");
+                                let snapshot_path = snapshot_pathbuf.as_path();
 
-                            let request_file = File::open(path.join("request.json"))
-                                .map_err(Error::CannotOpenSnapshotFile)?;
-                            let request = serde_json::from_reader(request_file)?;
+                                let request_file = File::open(path.join("request.json"))
+                                    .map_err(Error::CannotOpenSnapshotFile)?;
+                                let request = serde_json::from_reader(request_file)?;
 
-                            let response = f(request).await?;
+                                let response = f(request).await?;
 
-                            snapshot_test(snapshot_path, &response)
+                                snapshot_test(snapshot_path, &response)
+                            }
                         }
-                    }
-                );
+                    );
+                }
             }
         }
         Err(e) => println!("Warning: a snapshot folder could not be found: {e}"),
     }
 }
 
+// Read directory and sort inner files by alphabet
+fn read_dir_sorted_by_name(snapshots_dir: PathBuf) -> std::io::Result<Vec<std::fs::DirEntry>> {
+    let mut paths: Vec<_> = std::fs::read_dir(snapshots_dir)?
+        .map(|r| r.expect("Error reading snapshot directory entry"))
+        .collect();
+    paths.sort_by_key(std::fs::DirEntry::path);
```

Comment:
```suggestion
    let mut paths: Vec<_> = std::fs::read_dir(snapshots_dir)?.collect()?;
    paths.sort_by_key(std::fs::DirEntry::path);
```

I think this ought to work?

Reply:
No, it doesn't work. There are nested `Result` in `DirEntry` array. We need to unwrap twice
![2024-06-08_00-21](https://github.com/hasura/ndc-spec/assets/1615675/ea6eb85e-32a3-4cd2-85fe-04be80be0092)

Reply:
Oh, but then we'd have to change this line:
```
Warning: a snapshot folder could not be found: {e}
```

Reply:
So what's your suggestion?
That error line is used to catch the `read_dir` function.

```rust
let mut paths: Vec<_> = std::fs::read_dir(snapshots_dir)?
```

Reply:
I just pushed a commit to use `exists()` for the warning, and to use `expect` once for everything else (which should be an error now)`.

Will merge as soon as CI finishes.

Reply:
@paf31 feel free to update my PR. It's late in my place and I'm going to bed